### ISSUE TITLE
[BUG] - Deprecated aliases for numpy dtypes (after numpy v1.20)

### ISSUE THREAD

**Issue by jlubbersgeo** [morganjwilliams/pyrolite#88]
**Describe the bug**
Trying to use various `pyrolite.pyrochem` REE functions with numpy > 1.20 seems to not work. Example here is from the docs on using lambda functions.

**To Reproduce**

```python
# this example is from the docs

import pyrolite.plot
from pyrolite.util.synthetic import example_spider_data

df = example_spider_data(
    noise_level=0.05,
    size=100,
    start="DMM_WH2005",
    norm_to="Chondrite_PON",
    offsets={"Eu": 0.2},
).pyrochem.REE.pyrochem.denormalize_from("Chondrite_PON")
df.head(2)
```

**Expected behavior**
Expect dataframe to be displayed like in the [docs](https://pyrolite.readthedocs.io/en/develop/examples/geochem/lambdas.html). 

The type of error displayed is:
```
AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Environment:**
 - OS: Windows
 - Python Version: 3.10.11
 - Pyrolite Version 0.3.2
 - Numpy Version = 1.24.3

**Additional Context**
This seems to be just a package version issue?

**Comment by morganjwilliams** [comment]
Hey @jlubbersgeo, this indeed should just be a package version issue (i.e. it's fixed for newer versions of `pyrolite`, and could be managed for this example by pinning to an earlier `numpy` version). All of these references to the alias `np.float` should now have been replaced with either `float` for the generic case or `np.float64` for more specific cases (e.g. 710745aa3493bc8853e285f54824b8395003ff34 dealt with most of them).

**Comment by jlubbersgeo** [comment]
Well, here I was thinking I had the most up to date version of things! Turns out the --upgrade flag didn't really do that but an uninstall --> reinstall fixed the issue and now I'm on 3.4. Thanks for entertaining this silly request!

**Comment by morganjwilliams** [comment]
No worries, installation of `<anything>` tends to be the bane of my existence. Upgrading can indeed be worse. Unless you have some specific (e.g. environment/pinned version) requirements which might require some patching of old versions, I might close this one for now - but good to have on the books for folks googling/searching GitHub for answers!
